charts/grafana/templates/configmap.yaml

```diff
@@ -56,6 +56,14 @@ data:
   {{- end -}}
 {{- end -}}
 
+{{- if .Values.alerting }}
+{{ $root := . }}
+  {{- range $key, $value := .Values.alerting }}
+  {{ $key }}: |
+{{ tpl (toYaml $value | indent 4) $root }}
```

Comment:
Sorry, I was not finished testing this. I now have: `{{- tpl (toYaml $value) $root | nindent 4 }}` I still need to use the 'toYaml' as tpl expects a string (at least what I saw on testing and read).
And this produces now:
```yaml
data:
  ...
  alerts.yaml: |
    apiVersion: 1
    groups:
    ...
```

Reply:
@mmusenbr 
- `|` will be detected $value as a string (e.g. `alerts.yaml: |`).
- Using `nindent` will generate extra empty line between alert.yaml and its following content. For example:

``` yaml
  alerts.yaml: |

    apiVersion: 1
    groups:
    ...
```
- Dunno what version of your helm, my helm is `v.3.9.0`.

Reply:
@zanhsieh 
Ah thanks for the `|` treat as string info. Sheds some light.
Yes, this will generate an extra line, but I had, as at some point suggested, `{{-` in front of the template, which again consumes it from nintend.
So it is either `{{ tpl $value $root | indent 4 }}` or `{{- tpl $value $root | nindent 4 }}`, or? I will push the first version, as it matches the last suggestion I think and is closer to the way it is done in the other parts of the `configmap.yaml`.

And I hope this is the final point, should we use (on which I am, with my currently limited template knowledge not sure if it is the same or not):
```yaml
{{- if .Values.alerting }}
{{ $root := . }}
...
{{ tpl $value $root | indent 4 }}
....
```
or, as suggested at another point:
```yaml
{{- if .Values.alerting }}
...
{{ tpl $value $ | indent 4 }}
....
```